frontend/src/components/StakeSelectorItem.vue

```diff
@@ -82,136 +147,663 @@ export default {
     'rewardsDuration',
     'deprecated',
     'rewardDistributionTimeLeft',
-    'currentRewardEarned'
+    'currentRewardEarned',
+    'totalStaked',
+    'walletBalance',
   ],
+  components:{
+    Multiselect,
+  },
+  data() {
+    return{
+      textAmount: '',
+      idsToStake: [],
+      isOpen: false,
+      isDeposit: true,
+      loading: false,
+      errorWhenUpdating: null,
+      rewardClaimLoading: false,
+      stakeUnlockTimeLeftCurrentEstimate: 0,
+      stakeRewardDistributionTimeLeftCurrentEstimate: 0,
+      ownedLandIds: [],
+      stakedIds: []
+    };
+  },
+  async mounted() {
+    this.stakeUnlockTimeLeftCurrentEstimate = this.unlockTimeLeftInternal;
+    this.stakeRewardDistributionTimeLeftCurrentEstimate = this.rewardDistributionTimeLeftInternal;
+
+    this.stakeRewardProgressInterval = setInterval(async () => {
+      await this.fetchStakeDetails({ stakeType: this.stakeType });
+    }, 10 * 1000);
+
+    this.secondsInterval = setInterval(() => {
+      this.updateEstimates();
+    }, 1000);
```

Comment:
Are these something that is required to be updated so often?

Reply:
Not really, but seeing the countdown in the progressbar is good for the user i think. updateEstimates() is only decrementing two vars - I think that's fine.